src/Workspaces/Remote/ServiceHub/Host/ChecksumSynchronizer.cs

```diff
@@ -41,27 +41,27 @@ public async ValueTask SynchronizeSolutionAssetsAsync(Checksum solutionChecksum,
 
                 // first, get solution checksum object for the given solution checksum
                 var solutionCompilationChecksumObject = await _assetProvider.GetAssetAsync<SolutionCompilationStateChecksums>(
-                    assetHint: AssetHint.None, solutionChecksum, cancellationToken).ConfigureAwait(false);
+                    assetHint: AssetHint.SolutionOnly, solutionChecksum, cancellationToken).ConfigureAwait(false);
                 solutionChecksumObject = await _assetProvider.GetAssetAsync<SolutionStateChecksums>(
-                    assetHint: AssetHint.None, solutionCompilationChecksumObject.SolutionState, cancellationToken).ConfigureAwait(false);
+                    assetHint: AssetHint.SolutionOnly, solutionCompilationChecksumObject.SolutionState, cancellationToken).ConfigureAwait(false);
 
                 // second, get direct children of the solution
                 {
                     using var _ = PooledHashSet<Checksum>.GetInstance(out var checksums);
 
                     checksums.Add(solutionCompilationChecksumObject.SourceGeneratorExecutionVersionMap);
                     solutionChecksumObject.AddAllTo(checksums);
-                    await _assetProvider.SynchronizeAssetsAsync(assetHint: AssetHint.None, checksums, results: null, cancellationToken).ConfigureAwait(false);
+                    await _assetProvider.SynchronizeAssetsAsync(assetHint: AssetHint.SolutionOnly, checksums, results: null, cancellationToken).ConfigureAwait(false);
                 }
             }
 
             // third and last get direct children for all projects and documents in the solution 
-            foreach (var (projectChecksum, _) in solutionChecksumObject.Projects)
+            foreach (var (projectChecksum, projectId) in solutionChecksumObject.Projects)
             {
                 // These GetAssetAsync calls should be fast since they were just retrieved above.  There's a small
                 // chance the asset-cache GC pass may have cleaned them up, but that should be exceedingly rare.
                 var projectStateChecksums = await _assetProvider.GetAssetAsync<ProjectStateChecksums>(
-                    assetHint: AssetHint.None, projectChecksum, cancellationToken).ConfigureAwait(false);
+                    assetHint: projectId, projectChecksum, cancellationToken).ConfigureAwait(false);
```

Comment:
this is better than before.  now when syncing a project, we properly pass hte hint for it, so we don't have to sweep all projects looking for it.